src/textures/TextTextureRendererAdvanced.mjs

```diff
@@ -202,12 +204,36 @@ export default class TextTextureRendererAdvanced {
         }
         renderInfo.lineNum = lineNo + 1;
 
+        if (this._settings.h) {
+            renderInfo.h = this._settings.h;
+        } else if (renderInfo.maxLines && renderInfo.maxLines < renderInfo.lineNum) {
+            renderInfo.h = renderInfo.maxLines * renderInfo.lineHeight + fontSize / 2;
+        } else if (renderInfo.lineHeight > fontSize) {
+            // When lineheight is larger than the font size we're rendering, we set the height of the canvas based on the number of lines we're rendering.
+            // This makes each "line" a containing box that is line height sized, and text is positioned inside that box.
+            //
+            // Ideographic fonts may break this model, and require additional space?
+            renderInfo.h = renderInfo.lineNum * renderInfo.lineHeight
+        } else {
+            renderInfo.h = renderInfo.lineNum * renderInfo.lineHeight + fontSize / 2;
+        }
+
+        // This calculates the baseline offset in pixels from the font size.
+        // To retrieve this ratio, you would do this calculation:
+        //     (FontUnitsPerEm − hhea.Ascender − hhea.Descender) / (2 × FontUnitsPerEm)
+        //
+        // This give you the ratio for the baseline, which is then used to figure out 
+        // where the baseline is relative to the bottom of the text bounding box.
+        const baselineOffsetInPx = renderInfo.fontBaselineRatio * renderInfo.fontSize
```

Comment:
I took care here to make sure if no value was given here that the render will not have changed. Since a zero here will result in a zero overall change to the calculations for both `top` and `middle` vertical alignment.